pdc_client/config.py

```diff
@@ -0,0 +1,151 @@
+#
+# Copyright (c) 2018 Red Hat
+# Licensed under The MIT License (MIT)
+# http://opensource.org/licenses/MIT
+#
+
+import json
+import logging
+import os
+
+CONFIG_URL_KEY_NAME = 'host'
+CONFIG_INSECURE_KEY_NAME = 'insecure'
+CONFIG_SSL_VERIFY_KEY_NAME = 'ssl-verify'
+CONFIG_DEVELOP_KEY_NAME = 'develop'
+CONFIG_TOKEN_KEY_NAME = 'token'
+
+logger = logging.getLogger(__name__)
+
+
+class ServerConfigError(Exception):
+    """ Base class for ServerConfiguration exceptions. """
+    pass
+
+
+class ServerConfigNotFoundError(ServerConfigError):
+    """ Server configuration is missing. """
+    pass
+
+
+class ServerConfigMissingUrlError(ServerConfigError):
+    """ Server configuration is missing URL. """
+    pass
+
+
+class ServerConfigConflictError(ServerConfigError):
+    """ Same server defined in multiple config files. """
+    pass
+
+
+def _read_config_dir(file_path):
+    # get all json files in config directory and merge them
+    data = {}
+    if os.path.isdir(file_path):
+        files_list = os.listdir(file_path)
+        for file_name in files_list:
+            if file_name.endswith('.json'):
+                file_abspath = os.path.join(file_path, file_name)
+                with open(file_abspath, 'r') as config_file:
+                    config_dict = json.load(config_file)
+                    same_key = set(data.keys()) & set(config_dict.keys())
+                    if same_key:
+                        message = "'{}' keys existed in both {} config files".format(
+                            same_key, files_list)
+                        raise ServerConfigConflictError(message)
+
+                    data.update(config_dict)
+    return data
+
+
+def _read_config_file(file_path):
+    data = {}
+    if os.path.isfile(file_path):
+        with open(file_path, 'r') as config_file:
+            data = json.load(config_file)
+    return data
+
+
+def _default_server_configuration(server):
+    if "://" not in server:
+        message = "Failed to find configuration for server \"{}\"".format(server)
+        raise ServerConfigNotFoundError(message)
+
+    server_config = dict([(CONFIG_URL_KEY_NAME, server)])
```

Comment:
Is `{CONFIG_URL_KEY_NAME: server}` more readable?

Reply:
Yes, indeed. I thought that's not possible in Python - I must be thinking about different language.